lib/libzpool/uzfs_io.c

```diff
@@ -82,6 +84,10 @@ uzfs_write_data(zvol_state_t *zv, char *buf, uint64_t offset, uint64_t len,
 	uint64_t orig_offset = offset;
 	char *mdata = NULL, *tmdata = NULL, *tmdataend = NULL;
 
+	if (uzfs_write_size) {
```

Comment:
need to handle uzfs_write_size being 0

Reply:
already handled, in that case it will use zv->zv_volblocksize.

Reply:
I don't understand.. or some files need to be added to review?

Reply:
I'm considering the case where env is not provided, and uzfs_write_size will be 0

Reply:
got to clear my mis-understanding from you that this code will be executed when uzfs_write_size is non-zero. This check is fine